tls/s2n_server_new_session_ticket.c

```diff
@@ -76,6 +87,31 @@ int s2n_server_nst_send(struct s2n_connection *conn)
     return 0;
 }
 
+S2N_RESULT s2n_tls13_server_nst_send(struct s2n_connection *conn, s2n_blocked_status *blocked)
+{
+    ENSURE_REF(conn);
+
+    if (conn->mode != S2N_SERVER || conn->actual_protocol_version < S2N_TLS13) {
+        return S2N_RESULT_OK;
+    }
+
+    ENSURE(conn->tickets_sent <= conn->tickets_to_send, S2N_ERR_INTEGER_OVERFLOW);
+    while (conn->tickets_to_send - conn->tickets_sent > 0) {
+        uint8_t nst_data[TLS13_MAX_NEW_SESSION_TICKET_SIZE] = { 0 };
+        struct s2n_blob nst_blob = { 0 };
+        struct s2n_stuffer nst_stuffer = { 0 };
+        GUARD_AS_RESULT(s2n_blob_init(&nst_blob, nst_data, sizeof(nst_data)));
+        GUARD_AS_RESULT(s2n_stuffer_init(&nst_stuffer, &nst_blob));
```

Comment:
Instead of this, could we use handshake.io?

The issue with using handshake.io is that there's only one of it so we'd need to make sure that both s2n_send and s2n_recv don't try to use it at the same time, but NewSessionTickets are only sent by the server and nothing else should be using handshake.io after the handshake. It would save you the calculation of TLS13_MAX_NEW_SESSION_TICKET_SIZE, but it also seems like it introduces a future opportunity for strange concurrency bugs if we someday decide to use handshake.io for another post-handshake message :\

Reply:
Actually, we resize handshake.io at the end of negotiate. That wouldn't save us anything over just allocating some new memory :\